algopy_testing/tests/primitives/test_biguint.py

```diff
@@ -0,0 +1,740 @@
+        "ge",
+    ],
+)
+def test_biguint_comparison(get_avm_result: AVMInvoker, op_name: str, a: int, b: int) -> None:
+    avm_result = _get_avm_result(get_avm_result, f"verify_biguint_{op_name}", a, b)[1]
+    op = getattr(operator, op_name)
+
+    assert avm_result == op(BigUInt(a), BigUInt(b))
+    assert avm_result == op(BigUInt(a), b)
+    assert avm_result == op(a, BigUInt(b))
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [
+        (0, UInt64(0)),
+        (0, UInt64(1)),
+        (0, UInt64(MAX_UINT64)),
+        (1, UInt64(0)),
+        (1, UInt64(1)),
+        (1, UInt64(MAX_UINT64)),
+        (MAX_UINT512, UInt64(MAX_UINT64)),
+        (MAX_UINT64, UInt64(MAX_UINT64)),
+    ],
+)
+@pytest.mark.parametrize(
+    "op_name",
+    [
+        "eq",
+        "ne",
+        "lt",
+        "le",
+        "gt",
+        "ge",
+    ],
+)
+def test_biguint_comparison_uint64(
+    get_avm_result: AVMInvoker, op_name: str, a: int, b: UInt64
+) -> None:
+    avm_result = _get_avm_result(get_avm_result, f"verify_biguint_{op_name}_uint64", a, b)[1]
+    op = getattr(operator, op_name)
+    assert avm_result == op(BigUInt(a), b)
+    if a <= MAX_UINT64:
+        assert avm_result == op(UInt64(a), BigUInt(b))
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [
+        (0, 0),
+        (0, MAX_UINT512),
+        (MAX_UINT512, 0),
+        (1, 0),
+        (0, 1),
+        (1, 1),
+        (1, MAX_UINT512 - 1),
+        (MAX_UINT512 - 1, 1),
+    ],
+)
+def test_biguint_addition(get_avm_result: AVMInvoker, a: int, b: int) -> None:
+    avm_result = _get_avm_result(get_avm_result, "verify_biguint_add", a=a, b=b)
+    assert_biguint_and_backing_bytes(avm_result, BigUInt(a) + BigUInt(b))
+    assert_biguint_and_backing_bytes(avm_result, BigUInt(a) + b)
+    assert_biguint_and_backing_bytes(avm_result, a + BigUInt(b))
+    i = BigUInt(a)
+    i += b
+    assert_biguint_and_backing_bytes(avm_result, i)
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [
+        (0, UInt64(0)),
+        (0, UInt64(MAX_UINT64)),
+        (MAX_UINT512, UInt64(0)),
+        (1, UInt64(0)),
+        (0, UInt64(1)),
+        (1, UInt64(1)),
+        (10, UInt64(MAX_UINT64)),
+        (MAX_UINT512 - MAX_UINT64, UInt64(MAX_UINT64)),
+    ],
+)
+def test_biguint_addition_uint64(get_avm_result: AVMInvoker, a: int, b: UInt64) -> None:
+    avm_result = _get_avm_result(get_avm_result, "verify_biguint_add_uint64", a=a, b=b)
+    assert_biguint_and_backing_bytes(avm_result, BigUInt(a) + b)
+    assert_biguint_and_backing_bytes(avm_result, b + BigUInt(a))
+    i = BigUInt(a)
+    i += b
+    assert_biguint_and_backing_bytes(avm_result, i)
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [
+        (MAX_UINT512, 1),
+        (1, MAX_UINT512),
+        (MAX_UINT512, MAX_UINT512),
+    ],
+)
+def test_biguint_addition_result_overflow(get_avm_result: AVMInvoker, a: int, b: int) -> None:
+    avm_result = _get_avm_result(get_avm_result, "verify_biguint_add", a, b)[1]
```

Comment:
See my earlier comment on the implementation, but this test is revealing that currently our behaviour does not match AVM since AVM does not error on producing an overflow result, but our implementation does.